doc/developer_guide/psyir.rst

```diff
@@ -37,72 +37,147 @@ The PSyclone Internal Representation (PSyIR)
 ############################################
 
 The PSyclone Internal Representation (PSyIR) is a language-independent
-AST that PSyclone uses to represent the PSy layer and the kernel
-code. The PSyIR can be constructed from scratch or produced from
-existing code using one of the front-ends (Readers) and it can be
-transformed back to a particular language using the back-ends (Writers)
-provided in PSyclone.
-
-DataTypes
-=========
-
-PSyIR DataTypes currently support Scalar and Array types via the
-``ScalarType`` and ``ArrayType`` classes. However, they are designed
-to be easily extended. For example, a structure could be created thus:
+Intermediate Representation that PSyclone uses to represent the PSy (Parallel
```

Comment:
Isn't this already the third bullet point regarding extensibility? Or do you mean explain the difference between PSy-layer and kernel? Which in this case I am not sure it needs to be here as PSyIR just offers the options. It would be in a PSy-layer developers guide page that explains why it is implemented the way it is (which we don't have).

Reply:
Yes, I take your point. I think the problem is that I was thinking of the LFRic, GOcean and NEMO-specific extentions to PSyIR as being the PSyIR as that was what the PSyIR was until you added the kernel support. My confusion is partly down to you saying that PSyclone uses the PSyIR for the PSy-layer (but most of the PSy-layer is based on PSyIR domain-specific extensions). Perhaps you could add a sentence to the "It is extensible" bullet point saying that PSyclone extends it for the DSLs it supports?